core-web/libs/data-access/src/lib/dot-properties/dot-properties.service.ts

```diff
@@ -34,6 +39,13 @@ export class DotPropertiesService {
      * @memberof DotPropertiesService
      */
     getKeys(keys: string[]): Observable<Record<string, string>> {
+        if (this.featureConfig) {
+            const missingKeys = keys.filter((key) => !(key in (this.featureConfig || {})));
```

Comment:
filter method checks all the array and just returns the values that match the expression. It should be O(N) since it doesn't stop (or trim the array) until the end of the array.